docs/MSAL_2x_Migration_Guide.md

```diff
@@ -0,0 +1,553 @@
+
+We’re excited to announce the release of Microsoft Authentication Library (MSAL) 2.x for iOS and macOS platforms! This is a major release that introduces breaking changes, API cleanup, and a shift towards a more secure and standards-aligned SDK.
+
+This guide will help you:
+
+- Understand what changed in MSAL 2.x and why
+- Migrate your existing apps to MSAL 2.x
+- Avoid breaking your current authentication flow
+
+## MSAL 2.x vs 1.x: What Changed, Why It Matters, and How to Migrate
+
+### 1. Valid Redirect URI Format Requirement for Enterprise (AAD) Scenarios
+
+#### What Changed
+
+In MSAL 2.x, all enterprise (AAD) applications must specify a valid redirect URI in the format: `msauth.[BUNDLE_ID]://auth`.
+
+For applications migrating from ADAL, redirect URIs formatted as `<scheme>://[BUNDLE_ID]` remain valid and are still supported in MSAL 2.x.
+
+📖 For more information, see: [MSAL Redirect URI Format Requirements](https://learn.microsoft.com/en-us/entra/msal/objc/redirect-uris-ios#msal-redirect-uri-format-requirements)
+
+#### Why It Matters
+
+This standardization enables secure and valid redirection to brokered authentication with Microsoft Authenticator or Company Portal.
+
+#### How to Migrate
+
+##### 1. Register a valid redirect URI
+
+In the Azure Portal under-App Registrations > Authentication, configure a redirect URI in the format: `msauth.[BUNDLE_ID]://auth`.
+
+Note: If migrating from ADAL, the `<scheme>://[BUNDLE_ID]` format is still supported.
+
+⚠️ Important: Ensure this redirect URI is configured across all app targets and extensions (such as Share Extensions) to enable smooth brokered authentication.
+
+##### 2. Update Info.plist
+
+Add the following entry to your app’s Info.plist:
+
+```xml
+<key>CFBundleURLTypes</key>
+<array>
+  <dict>
+    <key>CFBundleURLSchemes</key>
+    <array>
+      <string>msauth.com.yourcompany.yourapp</string>
+    </array>
+  </dict>
+</array>
+```
+
+##### 3. Add URL schemes to Info.plist
+
+Include the following in Info.plist under LSApplicationQueriesSchemes:
+
+```xml
+<key>LSApplicationQueriesSchemes</key> 
+<array>  
+    <string>msauthv2</string>  
+    <string>msauthv3</string>  
+</array> 
+```
+
+##### 4. Initialize MSALPublicClientApplication using the configured redirect URI
+
+Objective-C:
+```objc
+MSALPublicClientApplicationConfig *config =  
+    [[MSALPublicClientApplicationConfig alloc] initWithClientId:@"your-client-id"  
+                                                     redirectUri:@"msauth.your.bundle.id://auth"  
+                                                     authority:authority]; 
+ 
+NSError *error = nil; 
+MSALPublicClientApplication *application =  
+    [[MSALPublicClientApplication alloc] initWithConfiguration:config error:&error]; 
+ 
+if (error) 
+{ 
+    NSLog(@"Error initializing MSAL: %@", error.localizedDescription); 
+     return; 
+}
+```
+
+Swift:
+```swift
+MSALPublicClientApplicationConfig *config =  
```

Comment:
This seems to be the same objective c code.  Please put the proper swift code

`let config = MSALPublicClientApplicationConfig(clientId: "your-client-id", redirectUri: "msauth.your.bundle.id://auth", authority: authority)`

Reply:
Good catch. 
Updated. Thanks